test/connectivity/modules/NetworkController/Settings.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace NetworkController
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+
+    class Settings
+    {
+        const string StartAfterPropertyName = "StartAfter";
+        const string DockerUriPropertyName = "DockerUri";
+        const string FrequencyPropertyName = "RunFrequencies";
+        const string NetworkIdPropertyName = "NetworkId";
+        const string NetworkControllerModePropertyName = "NetworkControllerMode";
+
+        static readonly Lazy<Settings> Setting = new Lazy<Settings>(
+            () =>
+            {
+                IConfiguration configuration = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("config/settings.json")
+                    .AddEnvironmentVariables()
+                    .Build();
+
+                var result = new List<Frequency>();
+                configuration.GetSection(FrequencyPropertyName).Bind(result);
+
+                return new Settings(
+                    configuration.GetValue<TimeSpan>(StartAfterPropertyName),
+                    configuration.GetValue<string>(DockerUriPropertyName),
+                    configuration.GetValue<string>(NetworkIdPropertyName),
+                    result,
+                    configuration.GetValue<NetworkControllerMode>(NetworkControllerModePropertyName));
+            });
+
+        Settings(
+            TimeSpan startAfter,
+            string dockerUri,
+            string networkId,
+            IList<Frequency> frequencies,
+            NetworkControllerMode mode)
+        {
+            this.StartAfter = startAfter;
+            this.Frequencies = frequencies;
+            this.DockerUri = Preconditions.CheckNonWhiteSpace(dockerUri, nameof(dockerUri));
+            this.NetworkId = Preconditions.CheckNonWhiteSpace(networkId, nameof(networkId));
+            this.NetworkControllerMode = mode;
+        }
+
+        public static Settings Current => Setting.Value;
+
+        public TimeSpan StartAfter { get; }
+
+        public IList<Frequency> Frequencies { get; }
```

Comment:
How do we plan on passing this to the module in the vsts pipeline? Stringified environment variable injected to the module? If so, sourcing this through the module twin's desired properties might be a better idea as you don't have to stringify..

Reply:
RunFrequencies__0__OfflineFrequency="00:00:10"

---
In reply to: [356447253](https://github.com/Azure/iotedge/pull/2040#discussion_r356447253) [](ancestors = 356447253)